packages/services/src/state_importer.rs

```diff
@@ -0,0 +1,179 @@
+use async_trait::async_trait;
+use ports::{
+    fuel::FuelBlock,
+    storage::Storage,
+    types::{StateFragment, StateSubmission},
+};
+
+use crate::{Result, Runner};
+
+pub struct StateImporter<Db, A> {
+    storage: Db,
+    fuel_adapter: A,
+}
+
+impl<Db, A> StateImporter<Db, A> {
+    pub fn new(storage: Db, fuel_adapter: A) -> Self {
+        Self {
+            storage,
+            fuel_adapter,
+        }
+    }
+}
+
+impl<Db, A> StateImporter<Db, A>
+where
+    Db: Storage,
+    A: ports::fuel::Api,
+{
+    async fn fetch_latest_block(&self) -> Result<FuelBlock> {
+        let latest_block = self.fuel_adapter.latest_block().await?;
+        // validate if needed
```

Comment:
we should definitely validate the block here. It should be straight forward though. We can do the same as in `block_committer`

Reply:
I've added the validation but I use the original block for now so that the data can be extracted later. There is quite some overlap between the importer and block_committer since it also import blocks.
We could turn that into a single component but I would wait for now until the state commitment process is better defined.